File: project_euler/Problem042/Problem042.py
import math

class Problem042:
    def solution(self) -> None:
        file_name, result = 'words.txt', 0
        with open(file_name, 'r') as handle:
            for line in handle:
                value = 0
                for letter in line.strip():
                    value += ord(letter) - ord('A') + 1
                auxiliary = math.sqrt((value << 3) + 1)
                if abs(math.floor(auxiliary + 0.5) - auxiliary) < 0.00000001 and (int(auxiliary) & 1):
                    result += 1
        print(result)

File: project_euler/Problem042/Problem042.go
package project_euler

import (
    "bufio"
    "fmt"
    "io"
    "math"
    "os"
)

func Problem042() {
    fileName := "words.txt"
    result := 0
    file, _ := os.Open(fileName)
    handle := bufio.NewReader(file)
    for true {
        line, _, err := handle.ReadLine()
        if io.EOF == err {
            break
        }

        value := 0
        for _, letter := range line {
            value += int(letter) - 'A' + 1
        }
        auxiliary := math.Sqrt(float64((value << 3) + 1))
        if math.Abs(math.Floor(auxiliary + 0.5) - auxiliary) < 0.00000001 && 0 != (int(auxiliary) & 1) {
            result++
        }
    }
    file.Close()
    fmt.Println(result)
}
